includes/class-boldgrid-inspirations-purchase-for-publish.php

```diff
@@ -1297,15 +1320,30 @@ public function asset_needs_purchase( $asset, $asset_type, $args = array() ) {
 		 * Is this image used within shortcode? For example, is image 123 used in a gallery, such as
 		 * [gallery ids='123,456'].
 		 */
-		$regexp = '[\[][^\]]+[\'\", ]' . $attachment_id . '[^0-9]+.*[\]]';
-		$query = '
-			SELECT `ID`
-			FROM ' . $wpdb->posts . '
-			WHERE	`post_status` IN ' . $post_status . ' AND
-					`post_type` IN ("page","post") AND
-					`post_content` REGEXP "' . $regexp . '"
-		';
-		$in_shortcode = $wpdb->get_var( $query );
+		$regexp = '[\\[][^\\]]+[\'\", ]' . $attachment_id . '[^0-9]+.*[\\]]';
+
+		/**
+		 * Previously, we were not using a prepare statement.
+		 * In order to ensure that the prepare statment
+		 * doesn't incorrectly escape the post status array,
+		 * we dynamically create the placeholders now.
+		 */
+		$placeholders = $this->get_prepare_placeholders(
+			array(
+				$post_status,
+				$regexp,
+			)
+		);
+
+		$in_shortcode = $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT `ID` FROM $wpdb->posts WHERE `post_status` IN ( " .
+				implode( ', ', array_fill( 0, count( $post_status ), '%s' ) ) .
+				" ) AND `post_type` IN ( 'page', 'post' ) AND `post_content` REGEXP %s",
+				...$placeholders
```

Comment:
Could you remove the `get_prepare_placeholders` method and all the lines that call it by simply changing...

```
...$placeholders
```

to...

```
$post_status,
$regexp
```

?

For me, it makes it easier to read by seeing the two variables we're passing in, rather than having to trace up the code and see what we're filling in `$placeholders`.

Not only do I have to know what's in `$placeholders`, but I also have to know how `get_prepare_placeholders` is manipulating the data.

It's just confusing me, complexity has been made higher without any obvious benefit.